### PR TITLE
{Upgrade} Skip upgrade if latest version is None

### PR DESCRIPTION
**Pull request by bebound**
**Related command**
`az upgrade`

**Description**

Currently, if it fails to get latest version from GitHub, latest version becomes `None`. It still prompts for upgrade.
image
This PR skips the upgrade if latest version is `None`.
image

**Testing Guide**

**History Notes**

[Component Name 1] BREAKING CHANGE: `az command a`: Make some customer-facing breaking change
[Component Name 2] `az command b`: Add some customer-facing feature

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [ ] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
Upgrade

File: src/azure-cli/azure/cli/command_modules/util/custom.py [hunk 1]
```diff
@@ -51,15 +51,17 @@ def upgrade_version(cmd, update_all=None, yes=None):  # pylint: disable=too-many
...
     update_cli = True
     from azure.cli.core.util import get_latest_from_github
-    try:
-        latest_version = get_latest_from_github()
-        if latest_version and parse(latest_version) <= parse(local_version):
-            logger.warning("You already have the latest azure-cli version: %s", local_version)
-            update_cli = False
-            if not update_all:
-                return
-    except Exception as ex:  # pylint: disable=broad-except
```
Comment (bebound) [inline src/azure-cli/azure/cli/command_modules/util/custom.py:61]: `get_latest_from_github` has cauch the exception, no need to catch again.

File: src/azure-cli/azure/cli/command_modules/util/custom.py [hunk 2]
```diff
@@ -51,15 +51,17 @@ def upgrade_version(cmd, update_all=None, yes=None):  # pylint: disable=too-many
...
-        logger.debug("Failed to get the latest version. %s", str(ex))
+    latest_version = get_latest_from_github()
```
Comment (jiasli) [inline src/azure-cli/azure/cli/command_modules/util/custom.py:54]: How to test it?

Comment (bebound) [reply src/azure-cli/azure/cli/command_modules/util/custom.py:54]: Disable Internet connection temporally.

**Review by jiasli — APPROVED** [review]

**Pull request merged**